Source/WebCore/editing/markup.h

```diff
@@ -99,10 +100,10 @@ WEBCORE_EXPORT String serializePreservingVisualAppearance(const SimpleRange&, Ve
 String serializePreservingVisualAppearance(const VisibleSelection&, ResolveURLs = ResolveURLs::No, SerializeComposedTree = SerializeComposedTree::No,
     IgnoreUserSelectNone = IgnoreUserSelectNone::Yes, PreserveBaseElement = PreserveBaseElement::No, Vector<Ref<Node>>* = nullptr);
 
+enum class IncludeShadowRoots : uint8_t { ExplicitOnly, Serializable, DeprecatedAll };
```

Comment:
I don't think `DeprecatedAll` is appropriate naming here. Nothing about this behavior is deprecated.
A crucial distinction is whether we want to serialize all shadow DOM for copy & paste purposes so I'd rather call it `Always` or `All`, or maybe `AllForCopy`.